includes/Modules/Analytics_4/GoogleAnalyticsAdmin/PropertiesEnhancedMeasurementService.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Class PropertiesEnhancedMeasurementService
+ *
+ * @package   Google\Site_Kit\Modules\Analytics_4\GoogleAnalyticsAdmin
+ * @copyright 2023 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics_4\GoogleAnalyticsAdmin;
+
+use Google\Site_Kit_Dependencies\Google\Service\GoogleAnalyticsAdmin;
+use Google\Site_Kit_Dependencies\Google_Client;
+
+/**
+ * Class for managing GA4 datastream enhanced measurement settings.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class PropertiesEnhancedMeasurementService extends GoogleAnalyticsAdmin {
+
+	/**
+	 * PropertiesEnhancedMeasurementResource instance.
+	 *
+	 * @var PropertiesEnhancedMeasurementResource
+	 */
+	public $properties_enhancedMeasurements; // phpcs:ignore WordPress.NamingConventions.ValidVariableName
```

Comment:
@techanvil I noticed this during review after it was already merged and perhaps it's inconsequential, but this is an example where the naming is inconsistent as was flagged in the original IBR (plural/singular).

Secondarily, the resource (datastream) is a sub-resource of the property but this not reflected. We should already have the class for `properties->datastreams`, so perhaps we can extend the class in the library instead? This way once the library is updated to a version which includes it, we wouldn't need to change its usage. See https://gist.github.com/aaemnnosttv/ede786f6caa7679cac981669c2f90cb5#file-googleanalyticsadmin-php-L1209

Reply:
> @techanvil I noticed this during review after it was already merged and perhaps it's inconsequential, but this is an example where the naming is inconsistent as was flagged in the original IBR (plural/singular).

Hi @aaemnnosttv, thanks for spotting this - this one slipped through the net in CR. AFAICT it's the only instance where we use the inconsistent plural naming. Tbh it's pretty marginal but I'm happy to create a followup issue to address if you think it's worthwhile doing so, please let me know.

> Secondarily, the resource (datastream) is a sub-resource of the property but this not reflected. We should already have the class for `properties->datastreams`, so perhaps we can extend the class in the library instead? This way once the library is updated to a version which includes it, we wouldn't need to change its usage. See https://gist.github.com/aaemnnosttv/ede786f6caa7679cac981669c2f90cb5#file-googleanalyticsadmin-php-L1209

Hmm, I'm a bit unclear what version your gist represents... In our current version, `v0.277.0`, the [`PropertiesDataStreams`](https://github.com/googleapis/google-api-php-client-services/blob/v0.277.0/src/GoogleAnalyticsAdmin/Resource/PropertiesDataStreams.php) class is used for the `v2alpha` API, and I'd be hesitant to reuse it for the `v1alpha` EM API methods that we're reintroducing here. I did consider extending [`PropertiesWebDataStreams`](https://github.com/googleapis/google-api-php-client-services/blob/v0.277.0/src/GoogleAnalyticsAdmin/Resource/PropertiesWebDataStreams.php), but that would mean we're bringing some additional methods in that support the `v1alpha` data stream  API & associated classes that we don't need.

I thought it would be cleaner to simply copy the stuff we do need from the old version of [`PropertiesWebDataStreams`](https://github.com/googleapis/google-api-php-client-services/blob/v0.220.0/src/GoogleAnalyticsAdmin/Resource/PropertiesWebDataStreams.php#L180-L186) instead into a new class. With these being `v1alpha` methods, I'm not sure how likely it would be for it to be reintroduced in a compatible manner in `PropertiesDataStreams` anyway, and I would imagine it's more likely we'll need to rework our implementation a bit when the beta API does provide support for EM.

That's my thinking here anyway, but I'm certainly open to suggestion if you want to take a different course with this.